Providers/nxOMSAutomationWorker/automationworker/worker/linuxutil.py

```diff
@@ -299,9 +297,58 @@ def get_cert_info(certificate_path):
 
     if p.poll() != 0:
         raise Exception("Unable to get certificate subject.")
-    subject = raw_subject.split("subject= ")[1].strip()
 
-    return issuer, subject, thumbprint
+    return parse_issuer_from_openssl_output(raw_issuer), \
+           parse_subject_from_openssl_output(raw_subject), \
+           parse_thumbprint_from_openssl_output(raw_fingerprint)
+
+
+def parse_thumbprint_from_openssl_output(raw_fingerprint):
+    """Parses the thumbprint value from the raw OpenSSL output.
+
+    Example output from openSSL:
+    SHA1 Fingerprint=3B:C3:70:46:00:0C:B2:0B:F9:86:98:CF:9D:11:DF:EB:22:B7:41:F5
+
+    Returns:
+        string : The certificate thumbprint.
+    """
+    return raw_fingerprint.split("SHA1 Fingerprint=")[1].replace(":", "").strip()
```

Comment:
Are you sure you'll always get the input you're expecting? You are risking an index out of bounds exception  (/equivalent) if you don't. Same comment applies to everywhere you've used split inline and applied an index without checking for size first.

Reply:
Good point. For now we are not going to make any changes (since we lost our distro test framework, we want to keep the behavior the same to make sure we do not break existing customers). Will make a note for @D1v38om83r  and I and we will address it once we have our infra back. Thanks @kjohn-msft